### PR TITLE
Fix animation startAt; add animation controls example

### PR DESCRIPTION
**Pull request by diarmidmackenzie**
Fix for #380 

- Fixes animation-mixer.startAt using THREE.AnimationAction.startAt.
- Adds an example that allows easy testing of all animation-mixer settings
- Documentation improvements.

### CODE REVIEW

File: examples/animation-controls/animation-controls.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+
+animationNames = {
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js]: You should use a `const` here, `const animationNames`

File: examples/animation-controls/animation-controls.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+updateAnimationMixer = () => {
+
+  const data = {}
+  data.clip = "none"
+  Object.entries(animationNames).forEach((name) => {
+
+    el = document.getElementById(name[0])
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js]: `const el`. If you don't declare a variable, it's a global variable.

File: examples/animation-controls/animation-controls.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  inputs.forEach((input) => {
+    input.addEventListener("change", updateAnimationMixer)
+  })
+
+  const selectors = document.querySelectorAll("select")
+
+  selectors.forEach((selector) => {
+    selector.addEventListener("change", updateAnimationMixer)
+  })
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js:59]: If you do the same for both input and select you can use `document.querySelectorAll("input,select")` and do the forEach once.

File: examples/animation-controls/animation-controls.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    data[key] = value
+  }
+  getValue("duration")
+  getValue("clampWhenFinished")
+  getValue("crossFadeDuration")
+  getValue("loop")
+  getValue("repetitions")
+  getValue("timeScale")
+  getValue("startAt")
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js]: I'm not super fan of this code. The name getValue is strange here for a function that doesn't return anything but set the value in a data object in the scope outside of the function.
Please consider something like that, I find it less confusing:
```js
const fieldKeys = ["duration", "clampWhenFinished", "crossFadeDuration", "loop", "repetitions", "timeScale", "startAt"]; // outside of updateAnimationMixer, no need to create the array every time you change the form.

fieldKeys.forEach((key) => {
    const value = document.getElementById(key).value

    if (AFRAME.components['animation-mixer'].schema[key].type === 'number' && isNaN(value)) {
      return;
    }
    data[key] = value
})
```

File: examples/animation-controls/animation-controls.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  idle: 'Armature|TRex_Idle',
+  jump: 'Armature|TRex_Jump',
+  run: 'Armature|TRex_Run',
+  walk: 'Armature|TRex_Walk',
+};
+
+
+
+updateAnimationMixer = () => {
+
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js:12]: Nit pick :) What coding style is this? You put a blank line at the top of the function but not the bottom? But you don't do it every time, so the style is not consistent. I'll personally don't put a blank line at the beginning of a function.
I know the threejs coding style that want a blank line at the beginning and at the end, but you don't use this style here.

**Review by vincentfretin — COMMENTED** [review]
Thank you again for investigating the issue and contributing an example!
I left some notes about the way you synchronize the form and the component.

**Comment by diarmidmackenzie** [comment]
Thanks for the feedback.  Sorry, I wrote the example code in a hurry, as I was mostly focussed on what the actual component did.  Should have tidied it up a lot more before submitting a PR.

Just doing some testing now, I spotted a couple of things I hadn't noticed before, which I want to investigate before merging this PR...

1. Duration doesn't work as I expected - it seems to slow the animation down (like timeScale, but in the reverse sense)

2. ClampWhenFinished seems to act as if it is always "true".

**Comment by diarmidmackenzie** [comment]
ClampWhenFInished was a bug in my code - for a checkbox, don't read "value" property, read "checked" instead.  Also updated on both click & change.

**Comment by diarmidmackenzie** [comment]
OK, I'd misunderstood the functon provided by 'duration'.

It is supposed to specify the duration of a singe loop of the animation.  That's always been the case, evidence [here](https://github.com/c-frame/aframe-extras/issues/278), for example.

So why does timeScale also exist?  Motivation seems to have been to provide a way [to pause an animation](https://github.com/c-frame/aframe-extras/issues/222).

I don't much like the interface.  Seems unnecessary to have two different properties to do pretty much exactly the same thing.  But we should maintain both properties for backwards compatibility reasons.

So for now, just documenting more clearly what 'duration' actually does.

File: examples/animation-controls/animation-controls.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const keys = ["duration",
+                "clampWhenFinished",
+                "crossFadeDuration",
+                "loop",
+                "repetitions",
+                "timeScale",
+                "startAt"]
+  keys.forEach((key) => {
+    const el = document.getElementById(key)
+    var value = el.value
```
Comment (vincentfretin) [inline examples/animation-controls/animation-controls.js]: For new code where you use `const`/`let`, please don't use `var`. Replace the `var` by `let` here.
For strings, can you please use single quotes or double quotes everywhere, and not a mix of the two? It seems we mainly use single quotes in this repo.

Comment (vincentfretin) [reply examples/animation-controls/animation-controls.js]: @diarmidmackenzie there was just this one last comment above and you can merge.

Comment (diarmidmackenzie) [reply examples/animation-controls/animation-controls.js]: Thanks for the reminder on this.  Will make the fix and merge soon.

**Pull request merged**